hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogImpl.java

```diff
@@ -360,6 +366,17 @@ public DatanodeDeletedBlockTransactions getTransactions(
             transactions.getBlocksDeleted() < blockDeletionLimit) {
           Table.KeyValue<Long, DeletedBlocksTransaction> keyValue = iter.next();
           DeletedBlocksTransaction txn = keyValue.getValue();
+
+          if (nextProcessTransactionId == keyValue.getKey() && !firstTime) {
+            // Before already processed this transactionId in the current iteration.
+            // Table is completely iterated once
+            nextProcessTransactionId = -1;
+            break;
```

Comment:
If the `nextProcessTransactionId` is removed form the table, we will not break here. This will end up adding duplicate entries into the transaction.